### PR TITLE
Mongo updates

### PR DESCRIPTION
**Pull request by bmacnaughton**
- handle old `Group_Reduce` KV source in addition to new.
- add `aggregate` `QueryOp` (stop synthesizing `count`)
- handle class and object versions of `bindings.Reporter` (allowed testing with N-API bindings)
- minor binding error message changes

### CODE REVIEW

File: lib/probes/mongodb-core.js [hunk 1]
```diff
@@ -255,6 +255,8 @@ const dataMakers = [
       data.Group_Reduce = cmd.group.$reduce.toString()
     } else if (typeof cmd.group.$reduce === 'object') {
       data.Group_Reduce = cmd.group.$reduce.code.toString()
+    } else {
```
Comment (prettycoder) [inline lib/probes/mongodb-core.js]:
```
 if (typeof cmd.group.$reduce === 'function') {
      data.Group_Reduce = cmd.group.$reduce.toString()	    
    } 
```
this is now also captured in the else clause, it seems that the test for 'object' is sufficient to cover the 2 cases.

File: lib/probes/mongodb-core.js [hunk 2]
```diff
@@ -267,14 +269,8 @@ const dataMakers = [
...
-    const p = cmd.pipeline
-    if (p[0].$match && p[1].$group) {
-      const p1 = p[1].$group
-      if (p1._id === null && p1.n && p1.n.$sum === 1) {
-        data.QueryOp = 'count'
-        data.Query = JSON.stringify(p[0].$match)
-      }
-    }
+    data.QueryOp = 'aggregate'
+    data.Pipeline = JSON.stringify(cmd.pipeline)
```
Comment (prettycoder) [inline lib/probes/mongodb-core.js]: nice! I double-checked with the spec ;)

File: test/probes/mongodb.test.js [hunk 3]
```diff
@@ -154,7 +154,30 @@ function makeTests (db_host, host, isReplicaSet) {
...
+      server = new mongodb.ReplSet(servers, options)
+      const mongoClient = new MongoClient(server, {})
+
+      mongoClient.connect((err, _client) => {
+        ao.loggers.test.debug('mongoClient() connect callback', err)
+        if (err) {
+          console.log('error connecting', err)
+          return done(err)
+        }
+        ctx.server = server
```
Comment (prettycoder) [inline test/probes/mongodb.test.js]: where is `ctx` defined?

File: lib/probes/mongodb-core.js (hunk 2 above)
Comment (bmacnaughton) [reply lib/probes/mongodb-core.js]: I updated the spec to conform to this, so it's a little bit self-referential.

File: test/probes/mongodb.test.js (hunk 3 above)
Comment (bmacnaughton) [reply test/probes/mongodb.test.js]: line 90.

**Pull request merged**